lib/manageiq/appliance_console/postgres_admin.rb

```diff
@@ -320,6 +329,84 @@ class << self
     ensure
       File.delete(error_path) if File.exist?(error_path)
     end
+
+    private_class_method def self.prepare_restore(backup_type, dbname)
+      if application_connections?
+        message = "Database restore failed. Shut down all evmserverd processes before attempting a database restore"
+        ManageIQ::ApplianceConsole.logger.error(message)
+        raise message
+      end
+
+      disable_replication(dbname)
+
+      conn_count = connection_count(backup_type, dbname)
+      if conn_count > 1
+        message = "Database restore failed. #{conn_count - 1} connections remain to the database."
+        ManageIQ::ApplianceConsole.logger.error(message)
+        raise message
+      end
+    end
+
+    private_class_method def self.application_connections?
+      result = [{"count" => 0}]
+
+      with_pg_connection do |conn|
+        result = conn.exec("SELECT COUNT(pid) FROM pg_stat_activity WHERE application_name LIKE '%MIQ%'")
+      end
+
+      result[0]["count"].to_i > 0
+    end
+
+    private_class_method def self.disable_replication(dbname)
+      require 'pg/logical_replication'
```

Comment:
Do we have to include this gem in the gemspec now?

Reply:
Well, if you searched for it using the `pg/...`, then yeah, it wouldn't show up.  But yes, it is required, but I think the specs wouldn't have passed if I didn't have it. 😉